src/matrix/numpy-array.h

```diff
@@ -0,0 +1,92 @@
+// matrix/numpy-array.h
+
+// Copyright 2020   Mobvoi AI Lab, Beijing, China (author: Fangjun Kuang)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef KALDI_MATRIX_NUMPY_ARRAY_H_
+#define KALDI_MATRIX_NUMPY_ARRAY_H_ 1
+
+#include <iostream>
+#include <vector>
+
+#include "matrix/kaldi-matrix.h"
+#include "matrix/kaldi-vector.h"
+
+namespace kaldi {
+
+/// \addtogroup matrix_group
+/// @{
+
+/** NumpyArray for reading *.npy files.
+ *
+ * This class implements the format described at
+ * https://github.com/numpy/numpy/blob/master/numpy/lib/format.py
+ */
+template <typename Real>
+class NumpyArray {
+ public:
+  static constexpr const char* kMagicPrefix = "\x93NUMPY";
+  static constexpr int kMagicLen = 6;
+  // TODO(fangjun): We support only format 1.0 at present.
+  // Support for 2.0 and 3.0 can be added later when needed.
+  static constexpr int8_t kMajorVersion = 1;
+  static constexpr int8_t kMinorVersion = 0;
```

Comment:
`constexpr` static members are supported since C++17. Is it really necessary?

Reply:
Removed to an unnamed namespace.